pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/aws/upgrade/index.md

```diff
@@ -0,0 +1,161 @@
+terraform state pull | jq -r '.resources[] | select(.module != null) |select(.module|startswith("module.dcos.module.dcos-infrastructure.module.dcos-lb")) | select(.type=="aws_lb_listener")| . as $i | .instances[] | "terraform import \""+$i.module+"."+$i.type+"."+$i.name+"[\\\""+.attributes_flat.port+"\\\"]" +"\" \"" + .attributes_flat.arn + "\""' > import_listener.sh
+```
+
+In the next step we will drop the complete state for these load balancers. We must make sure that the previous commands ran successfully and stored the targetgroups and listeners.
+In our example this looks like this:
+```
+$ cat import_targetgroups.sh
+terraform import "module.dcos.module.dcos-infrastructure.module.dcos-lb.module.dcos-lb-masters.module.masters.aws_lb_target_group.targetgroup[\"80\"]" "arn:aws:elasticloadbalancing:us-east-1:123456789123:targetgroup/generic-dcos-ee-demo-tg-80/dd6330d44538d834"
+[...]
+$ cat import_listener.sh
+terraform import "module.dcos.module.dcos-infrastructure.module.dcos-lb.module.dcos-lb-masters.module.masters.aws_lb_listener.listeners[\"80\"]" "arn:aws:elasticloadbalancing:us-east-1:123456789123:listener/app/generic-dcos-ee-demo/1675988fcc2f675e/0e177f08d3e2ddec"
+[...]
+```
+Overall our files have these amount of command lines:
+```
+$ wc -l import_listener.sh import_targetgroups.sh
+      11 import_listener.sh
+      11 import_targetgroups.sh
+      22 total
+```
+The numbers might differ on your infrastructure if you use additional ports. But should be equal for both files.
+Before we import these resources again we will drop state for `aws_lb_target_group` `aws_lb_listener` and `aws_lb_target_group_attachment`
+```
+$ for addr in $(terraform state pull | jq -r '.resources[] | select(.module != null) |select(.module|startswith("module.dcos.module.dcos-infrastructure.module.dcos-lb")) | select(.type=="aws_lb_target_group" or .type=="aws_lb_listener" or .type=="aws_lb_target_group_attachment")| . as $i | .instances[] | $i.module+"."+$i.type+"."+$i.name+(if .index_key == null then "" else "["+.index_key|tostring+"]" end)'); do terraform state rm "${addr}";done
+```
+Now lets import the target groups and listeners again
+```
+$ bash -x import_targetgroups.sh
+$ bash -x import_listener.sh
+```
+After we sucessfully imported the target groups and listeners again we run a complete terraform apply
+```
+$ terraform apply
+```
+The Plan will show create statements for `aws_lb_target_group_attachment` this is completely ok. Make sure there is __no__ destroy in this statement. With the create statements terraform will realise that the attachment already exists and places this information in its state.
```

Comment:
Suggested rewording:

"The resulting plan should show create actions for the `aws_lb_target_group_attachment` module, but no others.  There should be __no__ destroy actions, if there are, then you should answer no to the prompt and review the above steps again (??? what do you do here if you see a destroy).   

After you enter yes to proceed, Terraform will not actually create new attachments but will detect the existing attachments and add them to its state."